### PR TITLE
Git hooks, formatting, linting

### PR DESCRIPTION
**Pull request by kozross**
This PR standardizes our linting and formatting: specifically, we now use `ormolu` for source code, `cabal-fmt` for Cabal files, and HLint hints (with a few disabled hints). We also include a git hook and CI to ensure that each commit follows these rules.

### CODE REVIEW

**Comment by kozross** [comment]
The git hooks part of this doesn't actually work, because changes to `.git` don't get commited for... reasons. Working on that, but the rest can still be merged.

**Comment by Kleidukos** [comment]
@kozross I may have something for you: https://www.viget.com/articles/two-ways-to-share-git-hooks-with-your-team/

**Comment by kozross** [comment]
@Kleidukos This is reasonable. I'll do that to fix it up, but I still feel our CI should be checking these things, so I'll add that too.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,25 +1,45 @@
...
+* The Cabal file is formatted with
+  [`cabal-fmt`](https://github.com/phadej/cabal-fmt)
+
+To ensure that you are using the git hook, run the following, once:
+
+```
+git config core.hooksPath .githooks
+```
+
+You can also use the provided `Makefile` by running `make init`.
```
Comment (Kleidukos) [inline CONTRIBUTING.md:33]: love it

File: README.md [hunk 2]
```diff
@@ -2,4 +2,4 @@
 
 > Native embedded bindings to the libsodium cryptography library 
 
-This library vendors libsodium so that you don't have to.
+This library vendors libsodium so that you don't have to.
```
Comment (Kleidukos) [inline README.md:5]: Extraneous space?

**Review by Kleidukos — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (kozross) [reply README.md:5]: I put that in to re-run the actions to check how quickly our linting and formatting runs 'hot' versus 'cold'. Can remove.

Comment (Kleidukos) [reply README.md:5]: ah great :)

**Pull request merged**